### PR TITLE
Loading bundles.yml at startup

### PR DESCRIPTION
**Pull request by tahmidefaz**
Loads bundles.yml at startup

### CODE REVIEW

File: controllers/controllers_test.go [hunk 1]
```diff
@@ -18,7 +20,16 @@ import (
 const DEFAULT_ORG_ID string = "4384938490324"
 const DEFAULT_ACCOUNT_NUMBER string = "540155"
 
-func testRequest(method string, path string, accnum string, orgid string, fakeCaller func(string, string) SubscriptionsResponse, fakeBundleInfo func() []Bundle) (*httptest.ResponseRecorder, map[string]EntitlementsSection, string) {
+// func TestMain(m *testing.M) {
+// 	bundleInfo = fakeBundleInfo()
+// }
```
Comment (kruai) [inline controllers/controllers_test.go]: Please remove the commented-out code, unless there's an important reason for it being here.

File: controllers/controllers_test.go [hunk 2]
```diff
@@ -1,6 +1,8 @@
 package controllers
 
 import (
+	"fmt"
```
Comment (kruai) [inline controllers/controllers_test.go]: Recommend removing fmt unless you use it for something other than print statements

File: controllers/controllers_test.go [hunk 3]
```diff
@@ -18,7 +20,16 @@ import (
 const DEFAULT_ORG_ID string = "4384938490324"
 const DEFAULT_ACCOUNT_NUMBER string = "540155"
 
-func testRequest(method string, path string, accnum string, orgid string, fakeCaller func(string, string) SubscriptionsResponse, fakeBundleInfo func() []Bundle) (*httptest.ResponseRecorder, map[string]EntitlementsSection, string) {
+// func TestMain(m *testing.M) {
+// 	bundleInfo = fakeBundleInfo()
+// }
+
+func init() {
+	fmt.Println("Opening in test")
```
Comment (kruai) [inline controllers/controllers_test.go]: Remove print statements

File: controllers/controllers_test.go [hunk 4]
```diff
@@ -35,7 +46,7 @@ func testRequest(method string, path string, accnum string, orgid string, fakeCa
 	req = req.WithContext(ctx)
 	rr := httptest.NewRecorder()
 
-	GetBundleInfo = fakeBundleInfo
+	//GetBundleInfo = fakeGetBundle
```
Comment (kruai) [inline controllers/controllers_test.go]: Does this commented line need to be here?

File: controllers/subscriptions.go [hunk 5]
```diff
@@ -3,6 +3,7 @@ package controllers
 import (
 	"crypto/tls"
 	"encoding/json"
+	"fmt"
```
Comment (kruai) [inline controllers/subscriptions.go]: Recommend removing fmt unless you use it for something other than print statements

File: controllers/subscriptions.go [hunk 6]
```diff
@@ -36,6 +43,7 @@ func getClient() *http.Client {
 
 // GetBundleInfo returns the bundle information fetched from the YAML
 var GetBundleInfo = func() []types.Bundle {
+	fmt.Println("Opening in Index")
```
Comment (kruai) [inline controllers/subscriptions.go]: Remove print statements

File: controllers/controllers_test.go (hunk 1 above)
Comment (tahmidefaz) [reply controllers/controllers_test.go]: @kruai Sorry, that was a WIP! :smile:

File: config/main.go [hunk 7]
```diff
@@ -93,7 +93,9 @@ func initialize() {
 	options.SetDefault(Keys.CertsFromEnv, false)
 	options.SetDefault(Keys.Port, "3000")
 	options.SetDefault(Keys.SubsHost, "https://subscription.api.redhat.com")
-	options.SetDefault(Keys.CaPath, "./resources/ca.crt")
+	options.SetDefault(Keys.CaPath, "../resources/ca.crt")
+	options.SetDefault(Keys.Cert, "../test_data/test.cert")
+	options.SetDefault(Keys.Key, "../test_data/test.key")
```
Comment (shdunning) [inline config/main.go]: i would add a comment here stating that the defaults here are really intended for our test cases

File: controllers/subscriptions.go [hunk 8]
```diff
@@ -22,6 +22,12 @@ type getter func(string) []string
 
 var cache = ccache.New(ccache.Configure().MaxSize(500).ItemsToPrune(50))
 
+var bundleInfo []types.Bundle
+
+func init() {
+	bundleInfo = GetBundleInfo(config.GetConfig().Options.GetString(config.Keys.BundleInfoYaml))
+}
```
Comment (shdunning) [inline controllers/subscriptions.go]: similarly, add a comment here that the purpose of this is to read the yaml file one time at process startup.

File: config/main.go (hunk 7 above)
Comment (tahmidefaz) [reply config/main.go]: done!

File: controllers/subscriptions.go (hunk 8 above)
Comment (tahmidefaz) [reply controllers/subscriptions.go]: :rocket:

File: controllers/subscriptions.go [hunk 9]
```diff
@@ -22,6 +22,13 @@ type getter func(string) []string
 
 var cache = ccache.New(ccache.Configure().MaxSize(500).ItemsToPrune(50))
 
+var bundleInfo []types.Bundle
+
+func init() {
+	// Reads the yaml file at process startup
+	bundleInfo = GetBundleInfo(config.GetConfig().Options.GetString(config.Keys.BundleInfoYaml))
```
Comment (jhjaggars) [inline controllers/subscriptions.go]: Is is valid to start the service if you end up with no bundles?

Comment (tahmidefaz) [reply controllers/subscriptions.go]: I was playing around with the code a bit. It's quite simple to have entitlements return something like `{"":{"is_entitled":true}}`, when `bundles.yml` is unavailable. However, I am not sure if I should pursue that. Definitely need some directions here... :disappointed:

Comment (tahmidefaz) [reply controllers/subscriptions.go]: Sorry, I probably approached it the wrong way 🤦‍♂ 
I think that's a very good question, and maybe something we can address in a future PR

**Pull request merged**